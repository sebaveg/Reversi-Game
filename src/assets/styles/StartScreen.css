.formWrapper {
  align-items: center;
  background-color: #282c34;
  color: white;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  justify-content: center;
  text-align: center;
  max-width: 90%;
}

.formWrapper h5 {
  margin: 1.5rem 0;
}

.formWrapper form h2 {
  margin: 0 0 .5rem 0;
}

.formWrapper form input {
  font-size: 1em;
  padding: 10px 15px;
  min-width: 300px;
  text-align: center;
}

.form-group {
  margin-bottom: 2rem;
}

.button {
  background-color: green;
  border: 1px solid green;
  border-radius: 5px;
  box-shadow: 0 0 10px -3px white;
  color: white;
  cursor: pointer;
  font-size: 1.7rem;
  font-weight: bold;
  margin-top: 2rem;
  padding: 10px 0;
  text-shadow: 1px 0 #282c34;
  width: 100%;
}

.button:hover {
  background-color: darkgreen;
}

.button a {
  color: white;
  display: block;
  text-decoration: none;
}

.button a:hover {
  text-decoration: none;
  color: white;
}