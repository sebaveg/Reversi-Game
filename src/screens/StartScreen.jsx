import React, { useState } from 'react';
import { connect } from 'react-redux';
import { setNamePlayers, setColorPlayers } from '../actions';

import '../assets/styles/StartScreen.css';

// Only this screen use hooks
const StartScreen = (props) => {
  const [players, setPlayers] = useState({
    namePlayerOne: 'Player One',
    namePlayerTwo: 'Player Two',
  });

  // get value between 0 and 1. If value > 0.5 is 50% probability
  const randomColorDisk = () => (Math.random() > 0.5 ? 'white' : 'black');

  // Every change on input update the local state of this component
  const handleInput = (event) => {
    setPlayers({
      ...players,
      [event.target.name]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    props.setNamePlayers(players);
    props.setColorPlayers(randomColorDisk());
    props.history.push('/game');
  };

  return (
    <section className="formWrapper">
      <h5>Enter the name of the players</h5>
      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <h2>Player one</h2>
          <input type="text" name="namePlayerOne" placeholder="Insert name for player one" maxLength="25" onChange={handleInput} />
        </div>
        <div className="form-group">
          <h2>Player two</h2>
          <input type="text" name="namePlayerTwo" placeholder="Insert name for player two" maxLength="25" onChange={handleInput} />
        </div>
        <button type="submit" className="button">Start</button>
      </form>
    </section>
  );
};

const mapDispatchToProps = {
  setNamePlayers,
  setColorPlayers,
};

export default connect(null, mapDispatchToProps)(StartScreen);
