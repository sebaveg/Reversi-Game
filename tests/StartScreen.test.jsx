/* eslint-disable no-undef */
import React from 'react';
import configureMockStore from 'redux-mock-store';

import StartScreen from '../src/screens/StartScreen';

jest.mock('react-redux');

const initialState = {
};

const mockStore = configureMockStore();
let store;
let wrapper;

describe('<StartScreen /> Container', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<StartScreen store={store} />);
  });

  it(' +++ capturing snapshot of home', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(' +++ should render one <section>', () => {
    expect(wrapper.find('section')).toHaveLength(1);
  });
  it(' +++ should render one <h5>', () => {
    expect(wrapper.find('h5')).toHaveLength(1);
  });
  it(' +++ should render one <form>', () => {
    expect(wrapper.find('form')).toHaveLength(1);
  });
  it(' +++ should render one <div>', () => {
    expect(wrapper.find('div')).toHaveLength(2);
  });
  it(' +++ should render one <h2>', () => {
    expect(wrapper.find('h2')).toHaveLength(2);
  });
  it(' +++ should render one <input name="namePlayerOne">', () => {
    expect(wrapper.find('input[name="namePlayerOne"]')).toHaveLength(1);
  });
  it(' +++ should render one <input name="namePlayerTwo">', () => {
    expect(wrapper.find('input[name="namePlayerTwo"]')).toHaveLength(1);
  });
  it(' +++ should render one <button>', () => {
    expect(wrapper.find('button')).toHaveLength(1);
  });
});

describe('<StartScreen /> Container Events', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(<StartScreen store={store} />);
  });
  it('User insert name player One', () => {
    wrapper.find('input[name="namePlayerOne"]').simulate('change', { target: { value: 'Player One' } });
    expect(wrapper.find('input[name="namePlayerOne"]').prop('value')).toEqual('');
  });
  it('User insert name player Two', () => {
    wrapper.find('input[name="namePlayerTwo"]').simulate('change', { target: { value: 'Player Two' } });
    expect(wrapper.find('input[name="namePlayerTwo"]').prop('value')).toEqual('');
  });
  it('Form submit', () => {
    wrapper.find('button').simulate('click', {
      preventDefault: () => {},
    });
  });
});
