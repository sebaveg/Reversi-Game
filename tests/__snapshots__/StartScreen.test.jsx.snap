// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartScreen /> Container  +++ capturing snapshot of home 1`] = `
<StartScreen
  setColorPlayers={[Function]}
  setNamePlayers={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
